python/cudf/cudf/_lib/lists.pyx

```diff
@@ -58,3 +59,21 @@ def explode_outer(Table tbl, int explode_column_idx, bool ignore_index=False):
         column_names=tbl._column_names,
         index_names=None if ignore_index else tbl._index_names
     )
+
+
+def extract_element(Column col, size_type index):
+    if not isinstance(col.dtype, ListDtype):
+        raise TypeError("col is not a list column.")
```

Comment:
We don't need to verify this since this is an internal Cython function and the only public exposure will be via the `Series.list` accessor which will have already validated the type for us.